apps/cpu-profile-summarizer/src/start.ts

```diff
@@ -0,0 +1,191 @@
+        directories.push(`${dir}/${entry.name}`);
+      }
+    }
+  }
+
+  return files;
+}
+
+/**
+ * Prints the usage information for the application.
+ */
+function printUsage(): void {
+  console.log('Usage: cpu-profile-summarizer --input DIR [[--input DIR] ...] --output TSV_FILE');
+}
+
+/**
+ * Processes a set of CPU profiles and aggregates the results.
+ * Uses a worker pool.
+ * @param profiles - The set of .cpuprofile files to process
+ * @returns A summary of the profiles
+ */
+async function processProfilesAsync(profiles: Set<string>): Promise<IProfileSummary> {
+  const maxWorkers: number = Math.min(profiles.size, os.availableParallelism());
+  console.log(`Processing ${profiles.size} profiles using ${maxWorkers} workers...`);
+  const workerPool: WorkerPool = new WorkerPool({
+    id: 'cpu-profile-summarizer',
+    maxWorkers,
+    workerScriptPath: path.resolve(__dirname, 'worker.js')
+  });
+
+  const summary: IProfileSummary = new Map();
+
+  let processed: number = 0;
+  await Promise.all(
+    Array.from(profiles, async (profile: string) => {
+      const worker: Worker = await workerPool.checkoutWorkerAsync(true);
+      const responsePromise: Promise<IMessageFromWorker[]> = once(worker, 'message');
+      worker.postMessage(profile);
+      const { 0: messageFromWorker } = await responsePromise;
+      if (messageFromWorker.type === 'error') {
+        console.error(`Error processing ${profile}: ${messageFromWorker.data}`);
+      } else {
+        ++processed;
+        console.log(`Processed ${profile} (${processed}/${profiles.size})`);
+        mergeProfileSummaries(summary, messageFromWorker.data);
+      }
+      workerPool.checkinWorker(worker);
+    })
+  );
+
+  await workerPool.finishAsync();
+
+  return summary;
+}
+
+function writeSummaryToTsv(tsvPath: string, summary: IProfileSummary): void {
+  const dir: string = path.dirname(tsvPath);
+  fs.mkdirSync(dir, { recursive: true });
+
+  let tsv: string = `Self Time (seconds)\tTotal Time (seconds)\tFunction Name\tURL\tLine\tColumn`;
+  for (const { selfTime, totalTime, functionName, url, lineNumber, columnNumber } of summary.values()) {
+    const selfSeconds: string = (selfTime / 1e6).toFixed(3);
+    const totalSeconds: string = (totalTime / 1e6).toFixed(3);
+
+    tsv += `\n${selfSeconds}\t${totalSeconds}\t${functionName}\t${url}\t${lineNumber}\t${columnNumber}`;
+  }
+
+  fs.writeFileSync(tsvPath, tsv, 'utf8');
+  console.log(`Wrote summary to ${tsvPath}`);
+}
+
+async function executeAsync(): Promise<boolean> {
+  const {
+    values: { input, output, help }
+  } = parseArgs({
+    args: process.argv.slice(2),
+    options: {
+      input: {
+        type: 'string',
+        alias: 'i',
+        multiple: true,
+        required: true
+      },
+      output: {
+        type: 'string',
+        alias: 'o'
+      },
+      help: {
+        type: 'boolean',
+        alias: 'h'
+      }
+    }
+  });
+
+  if (help) {
+    printUsage();
```

Comment:
Feels like this should be `ts-command-line` and the work to use the native arg parser should go in at that level instead. Feels like this just makes more work for us to migrate in the future.

May as well keep our CLI-based tools on the same stack